Observer/CaptureObserver.php

```diff
@@ -262,24 +263,31 @@ private function shippingTrackingInfo($invoice)
      * @param $orderObject
      * @param $payment
      * @param $storeCode
+     * @param $paymentType
      *
      * @throws ResponseHeaderException
      */
-    private function sendInvoiceRequest($invoice, $orderLines, $orderObject, $payment, $storeCode)
+    private function sendInvoiceRequest($paymentType, $invoice, $orderLines, $orderObject, $payment, $storeCode)
     {
-        $api = new CaptureReservation($this->systemConfig->getAuth($storeCode));
-        if ($invoice->getTransactionId()) {
-            $api->setInvoiceNumber($invoice->getTransactionId());
-        }
-        $api->setAmount((float)number_format($invoice->getGrandTotal(), 2, '.', ''));
-        $api->setOrderLines($orderLines);
-        $shippingTrackingInfo = $this->shippingTrackingInfo($invoice);
-        /*send shipping tracking info if exists*/
-        if (!empty($shippingTrackingInfo)) {
-            $api->setTrackingInfo($shippingTrackingInfo);
-        }
+        if ($paymentType === 'subscription') {
+            $api = new ChargeSubscription($this->systemConfig->getAuth($storeCode));
+            $api->setTransaction($payment->getLastTransId());
+            $api->setAmount((float)number_format($invoice->getGrandTotal(), 2, '.', ''));
```

Comment:
If you are casting the result of number_format (a `string`) to a `float` then you are doing something wrong.

```suggestion
            $api->setAmount(round($invoice->getGrandTotal(), 2));
```